src/Symfony/Component/HttpKernel/Controller/ControllerReference.php

```diff
@@ -26,20 +26,15 @@
  */
 class ControllerReference
 {
-    public array $attributes = [];
-    public array $query = [];
-
     /**
      * @param string $controller The controller name
      * @param array  $attributes An array of parameters to add to the Request attributes
      * @param array  $query      An array of parameters to add to the Request query string
      */
     public function __construct(
         public string $controller,
-        array $attributes = [],
-        array $query = [],
+        public array $attributes = [],
+        public array $query = [],
```

Comment:
Typically here we should be extra careful: this class is already using CPP but we excluded two properties from CPP to be on the safe side, by not removing default values.